pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1554,6 +1557,25 @@ private void createPartitionMetadataProvider(String reason) {
     _partitionMetadataProvider = _streamConsumerFactory.createPartitionMetadataProvider(_clientId, _partitionGroupId);
   }
 
+  /*
+   * Updates the ingestion delay if messages were processed using the time stamp for the last consumed event.
+   *
+   * @param indexedMessagesCount
+   */
+  private void updateIngestionDelay(int indexedMessageCount) {
+    if (_catchingUpPhase) {
```

Comment:
If this is `false` in the beginning, we also need to update the ingestion delay when it becomes `true`, or the lag will already be propagated.

Reply:
Code now changed to use server ready signal